editoast/src/views/test_app.rs

```diff
@@ -0,0 +1,169 @@
+        assert!(self.core_client.is_none());
+        self.core_client = Some(core_client);
+        self
+    }
+
+    pub fn default_app(
+    ) -> TestApp<impl Service<Request, Response = ServiceResponse<BoxBody>, Error = Error>> {
+        let pool = DbConnectionPoolV2::for_tests();
+        let core_client = CoreClient::default();
+        TestAppBuilder::new()
+            .db_pool(pool)
+            .core_client(core_client)
+            .build()
+    }
+
+    pub fn build(
+        self,
+    ) -> TestApp<impl Service<Request, Response = ServiceResponse<BoxBody>, Error = Error>> {
+        let json_cfg = JsonConfig::default()
+            .limit(250 * 1024 * 1024) // 250MB
+            .error_handler(|err, _| InternalError::from(err).into());
+
+        let redis = RedisClient::new(RedisConfig::default()).expect("cannot get redis client");
+
+        let core_client = Data::new(self.core_client.expect(
+            "No core client provided to TestAppBuilder, use Default or provide a core client",
+        ));
+        let ref_core_client = core_client.clone().into_inner();
+
+        let mut app = App::new()
+            .wrap(NormalizePath::trim())
+            .app_data(json_cfg)
+            .app_data(Data::new(redis))
+            .app_data(Data::new(CHashMap::<i64, InfraCache>::default()))
+            .app_data(Data::new(MapLayers::parse()))
+            .app_data(Data::new(MapLayersConfig::default()))
+            .app_data(core_client);
+
+        let ref_db_pool = if self.db_pool_v1 {
+            let config = PostgresConfig::default();
+            let pg_config_url = config.url().expect("cannot get postgres config url");
+            let pool = create_connection_pool(pg_config_url, config.pool_size)
+                .expect("could not create connection pool for tests");
+            app = app.app_data(Data::new(pool));
+            None
+        } else {
+            let pool = self.db_pool.expect("No database connection pool provided to TestAppBuilder, use Default or provide a connection pool");
+            let pool = Data::new(pool);
+            let ref_db_pool = pool.clone().into_inner();
+            app = app.app_data(pool);
+            Some(ref_db_pool)
+        };
+
+        let service =
+            futures::executor::block_on(init_service(app.service(crate::views::routes())));
+        TestApp {
+            service,
+            db_pool: ref_db_pool,
+            core_client: ref_core_client,
+        }
+    }
+}
+
+pub(crate) struct TestApp<S>
+where
+    S: Service<Request, Response = ServiceResponse<BoxBody>, Error = Error>,
+{
+    pub(crate) service: S,
+    /// A reference to the database connection pool
+    ///
+    /// The Option<> lasts while the pool V1 is still around.
+    db_pool: Option<Arc<DbConnectionPoolV2>>,
+    core_client: Arc<CoreClient>,
+}
+
+impl<S> TestApp<S>
+where
+    S: Service<Request, Response = ServiceResponse<BoxBody>, Error = Error>,
+{
+    #[allow(dead_code)]
```

Comment:
Why is this `dead_code`? The entire module is protected by `#[cfg(test)]` so I would expect all of these functions to be used in test code at least? If there is a real reason, maybe we need a comment explaining why we allow `dead_code` here.